src/rebar3_cargo_compile_prv.erl

```diff
@@ -154,8 +149,17 @@ write_header(App, NifLoadPaths) ->
         "-endif.\n"
     ],
 
-    OutDir = rebar_app_info:dir(App),
-    OutPath = filename:join([OutDir, "src", "crates.hrl"]),
+    OutDir0 = rebar_app_info:dir(App),
+    IncludeDir = filename:join(OutDir0, "include"),
+    SrcDir = filename:join(OutDir0, "src"),
+    % prefer the include dir to generate crates.hrl to (if it exists)
+    OutDir = case filelib:is_dir(IncludeDir) of
```

Comment:
No, the `crates.hrl` is not a "public" header file.

Reply:
agree, and it should be git ignored by any project using it. But if it's getting generated somewhere might as well be `include` (if possible), tipically only `.erl` files live in `src`

Reply:
The convention seems to be to have internal headers in `src`.

Reply:
ok, i can move this commit out of this PR context, another way would be to introduce an option for the target dir of `crates.hrl` that defaults to `src`, thoughts?

Reply:
7dad8595 has been removed

Reply:
1e489afe that contained this change has been dropped